ksql-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -131,18 +131,20 @@ private Router setupRouter() {
   }
 
   private static void handleFailure(final RoutingContext routingContext) {
+    // Avoid calling HttpServerResponse#setStatusCode() with a negative status code as that throws
+    final int statusCode = routingContext.statusCode() < 0 ? 500 : routingContext.statusCode();
```

Comment:
Out of interest.... where is the negative status code coming from?  What ever the source, surely that is a bug that needs fixing?

Reply:
The negative status code comes from the vert.x basic auth handler implementation, which fails the context with a throwable if it cannot decode the credentials: https://github.com/vert-x3/vertx-web/blob/f55625ef1a4174730ece8f9ea5a69632dc093280/vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BasicAuthHandlerImpl.java#L76
That method sets a status code of -1: https://github.com/vert-x3/vertx-web/blob/f55625ef1a4174730ece8f9ea5a69632dc093280/vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java#L167
Is this considered a bug? Seems intentional since all invocations of the method result in a status code of -1.

Reply:
It's expected it can be -1. If failure handler is called due to an unhandled exception from another handler or if routingContext.fail() is called without specifying a status code, it will be -1.